src/libImaging/Draw.c

```diff
@@ -988,7 +1165,7 @@ int
 ImagingDrawEllipse(Imaging im, int x0, int y0, int x1, int y1,
                    const void* ink, int fill, int width, int op)
 {
-    return ellipse(im, x0, y0, x1, y1, 0, 360, ink, fill, width, CHORD, op);
+    return ellipseNew(im, x0, y0, x1, y1, ink, fill, width, op);
```

Comment:
Is there any chance new implementation is used for other functions? At least for ImagingDrawChord?

Reply:
Yeah, but it would involve another layer of processing that clips ellipse down to arc.
I've got it more or less done locally, maybe I'll commit that later. The code is gonna look funky though due to all the floating point geometry and special cases processing...

Reply:
Arcs, chords and pies are _kinda_ working now.
At least compared to whatever was in master before.

I also added a test for various size ellipses. I added it long ago but forgot to push.